File: VB-Examples/09_Charts/GaugeChart.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls
Imports Spire.Xls.Charts

Namespace GaugeChart
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new workbook object
            Dim workbook As New Workbook()

            ' Get the first worksheet in the workbook and assign it to the "sheet" variable
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Set the name of the worksheet to "Gauge Chart"
            sheet.Name = "Gauge Chart"

            ' Call the function to create chart data on the sheet
            CreateChartData(sheet)

            ' Add a doughnut chart to the sheet and assign it to the "chart" variable
            Dim chart As Chart = sheet.Charts.Add(ExcelChartType.Doughnut)

            ' Set the data range for the chart to cells A1 to A5
            chart.DataRange = sheet.Range("A1:A5")

            ' Disable automatic series data detection from the range
            chart.SeriesDataFromRange = False

            ' Enable the legend for the chart
            chart.HasLegend = True

            ' Set the position of the chart within the worksheet
            chart.LeftColumn = 2
            chart.TopRow = 7
            chart.RightColumn = 9
            chart.BottomRow = 25

            ' Get the first series in the chart and assign it to the "cs1" variable
            Dim cs1 As ChartSerie = CType(chart.Series("Value"), ChartSerie)

            ' Set the size of the doughnut hole to 60% of the chart's diameter
            cs1.Format.Options.DoughnutHoleSize = 60

            ' Set the starting angle for the first slice of the doughnut to 270 degrees (anti-clockwise)
            cs1.DataFormat.Options.FirstSliceAngle = 270

            ' Set the fill color of the first data point in the series to yellow
            cs1.DataPoints(0).DataFormat.Fill.ForeColor = Color.Yellow

            ' Set the fill color of the second data point in the series to pale violet red
            cs1.DataPoints(1).DataFormat.Fill.ForeColor = Color.PaleVioletRed

            ' Set the fill color of the third data point in the series to dark violet
            cs1.DataPoints(2).DataFormat.Fill.ForeColor = Color.DarkViolet

            ' Make the fourth data point in the series invisible by setting its fill visibility to false
            cs1.DataPoints(3).DataFormat.Fill.Visible = False

            ' Add a new series of type "Pointer" (pie chart) to the chart and assign it to the "cs2" variable
            Dim cs2 As ChartSerie = CType(chart.Series.Add("Pointer", ExcelChartType.Pie), ChartSerie)

            ' Set the values for the second series from cells D2 to D4
            cs2.Values = sheet.Range("D2:D4")

            ' Use the secondary axis for the second series
            cs2.UsePrimaryAxis = False

            ' Enable data labels for the first data point in the second series
            cs2.DataPoints(0).DataLabels.HasValue = True

            ' Set the starting angle for the first slice of the second series to 270 degrees (anti-clockwise)
            cs2.DataFormat.Options.FirstSliceAngle = 270

            ' Make the first data point in the second series invisible by setting its fill visibility to false
            cs2.DataPoints(0).DataFormat.Fill.Visible = False

            ' Set the fill color of the second data point in the second series to black
            cs2.DataPoints(1).DataFormat.Fill.FillType = ShapeFillType.SolidColor
            cs2.DataPoints(1).DataFormat.Fill.ForeColor = Color.Black

            ' Make the third data point in the second series invisible by setting its fill visibility to false
            cs2.DataPoints(2).DataFormat.Fill.Visible = False

            ' Save the workbook to a file named "Output.xlsx" in the Excel 2010 format
            workbook.SaveToFile("Output.xlsx", FileFormat.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()
            ExcelDocViewer("Output.xlsx")
		End Sub

		Private Sub CreateChartData(ByVal sheet As Worksheet)
            ' Set the value of cell A1 to "Value"
            sheet.Range("A1").Value = "Value"

            ' Set the value of cell A2 to "30"
            sheet.Range("A2").Value = "30"

            ' Set the value of cell A3 to "60"
            sheet.Range("A3").Value = "60"

            ' Set the value of cell A4 to "90"
            sheet.Range("A4").Value = "90"

            ' Set the value of cell A5 to "180"
            sheet.Range("A5").Value = "180"

            ' Set the value of cell C2 to "value"
            sheet.Range("C2").Value = "value"

            ' Set the value of cell C3 to "pointer"
            sheet.Range("C3").Value = "pointer"

            ' Set the value of cell C4 to "End"
            sheet.Range("C4").Value = "End"

            ' Set the value of cell D2 to "10"
            sheet.Range("D2").Value = "10"

            ' Set the value of cell D3 to "1"
            sheet.Range("D3").Value = "1"

            ' Set the value of cell D4 to "189"
            sheet.Range("D4").Value = "189"
        End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace

File: CS-Examples/09_Charts/GaugeChart.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Xls;
using Spire.Xls.Charts;

namespace GaugeChart
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a Workbook
            Workbook workbook = new Workbook();

            // Get the first sheet and set its name
            Worksheet sheet = workbook.Worksheets[0];
            sheet.Name = "Gauge Chart";

            // Set chart data
            CreateChartData(sheet);

            // Add a Doughnut chart
            Chart chart = sheet.Charts.Add(ExcelChartType.Doughnut);
            chart.DataRange = sheet.Range["A1:A5"];
            chart.SeriesDataFromRange = false;
            chart.HasLegend = true;

            // Set the position of chart
            chart.LeftColumn = 2;
            chart.TopRow = 7;
            chart.RightColumn = 9;
            chart.BottomRow = 25;

            // Get the series 1
            ChartSerie cs1 = (ChartSerie)chart.Series["Value"];
            cs1.Format.Options.DoughnutHoleSize = 60;
            cs1.DataFormat.Options.FirstSliceAngle = 270;

            // Set the fill color
            cs1.DataPoints[0].DataFormat.Fill.ForeColor = Color.Yellow;
            cs1.DataPoints[1].DataFormat.Fill.ForeColor = Color.PaleVioletRed;
            cs1.DataPoints[2].DataFormat.Fill.ForeColor = Color.DarkViolet;
            cs1.DataPoints[3].DataFormat.Fill.Visible = false;

            // Add a series with pie chart
            ChartSerie cs2 = (ChartSerie)chart.Series.Add("Pointer", ExcelChartType.Pie);

            // Set the value
            cs2.Values = sheet.Range["D2:D4"];
            cs2.UsePrimaryAxis = false;
            cs2.DataPoints[0].DataLabels.HasValue = true;
            cs2.DataFormat.Options.FirstSliceAngle = 270;
            cs2.DataPoints[0].DataFormat.Fill.Visible = false;
            cs2.DataPoints[1].DataFormat.Fill.FillType = ShapeFillType.SolidColor;
            cs2.DataPoints[1].DataFormat.Fill.ForeColor = Color.Black;
            cs2.DataPoints[2].DataFormat.Fill.Visible = false;

            // Save the file
            workbook.SaveToFile("Output.xlsx", FileFormat.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            ExcelDocViewer("Output.xlsx");
		}

		private void CreateChartData(Worksheet sheet)
		{
            //Set value of specified cell
            sheet.Range["A1"].Value = "Value";
            sheet.Range["A2"].Value = "30";
            sheet.Range["A3"].Value = "60";
            sheet.Range["A4"].Value = "90";
            sheet.Range["A5"].Value = "180";
            sheet.Range["C2"].Value = "value";
            sheet.Range["C3"].Value = "pointer";
            sheet.Range["C4"].Value = "End";
            sheet.Range["D2"].Value = "10";
            sheet.Range["D3"].Value = "1";
            sheet.Range["D4"].Value = "189";
		}

		private void ExcelDocViewer( string fileName )
		{
			try
			{
				System.Diagnostics.Process.Start(fileName);
			}
			catch{}
		}

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}
